runtime/runtime_controller.cc

```diff
@@ -48,7 +48,9 @@ std::unique_ptr<RuntimeController> RuntimeController::Spawn(
     const std::function<void(int64_t)>& p_idle_notification_callback,
     const fml::closure& p_isolate_create_callback,
     const fml::closure& p_isolate_shutdown_callback,
-    std::shared_ptr<const fml::Mapping> p_persistent_isolate_data) const {
+    std::shared_ptr<const fml::Mapping> p_persistent_isolate_data,
+    fml::WeakPtr<IOManager> io_manager,
+    fml::WeakPtr<ImageDecoder> image_decoder) const {
```

Comment:
Instead of passing these 2 things down.  I think it makes more sense to just pass down the correct `UIDartState::Context` which can get custom created in `Engine::Spawn`.

Reply:
> 

Done!